app/controllers/exam_templates_controller.rb

```diff
@@ -73,6 +73,37 @@ def update
     redirect_to action: 'index'
   end
 
+  def create_template_division
+    assignment = Assignment.find(params[:assignment_id])
+    template = assignment.exam_templates.find_by(id: params[:id])
+    division_start = params[:create_division][:start]
+    division_end = params[:create_division][:end]
+    division_label = params[:create_division][:label]
+
+    new_template_division = template.template_divisions.create_with_associations(
+      assignment.id,
+      label: division_label,
+      start: division_start,
+      end: division_end
+    )
+    # sending flash message if saved
+    if new_template_division.save
+      flash_message(:success, t('template_divisions.create.success'))
+    else
+      flash_message(:error, t('template_divisions.create.failure'))
+    end
+    redirect_to action: 'index'
+  end
+
+  # Dialog to create template division.
+  def create_template_division_dialog
```

Comment:
If you rename this method and route to `create_division_dialog`, you won't have to include the explicit render at the bottom.